scipy/integrate/_ivp/base.py

```diff
@@ -15,6 +15,9 @@ def check_arguments(fun, y0, support_complex):
 
     if y0.ndim != 1:
         raise ValueError("`y0` must be 1-dimensional.")
+    
+    if not np.isfinite(y0).all():
+        raise ValueError("All components of the initial state `y0` must be finite.")
```

Comment:
For the matching regression test, this seems to hang on `main` and pass instantly here:

```diff
diff --git a/scipy/integrate/_ivp/tests/test_ivp.py b/scipy/integrate/_ivp/tests/test_ivp.py
index 2fb294823..0ed0d752a 100644
--- a/scipy/integrate/_ivp/tests/test_ivp.py
+++ b/scipy/integrate/_ivp/tests/test_ivp.py
@@ -1038,3 +1038,9 @@ def test_args_single_value():
 
     sol = solve_ivp(fun_with_arg, (0, 0.1), [1], args=(-1,))
     assert_allclose(sol.y[0, -1], np.exp(-0.1))
+
+
+@pytest.mark.parametrize("f0_fill", [np.nan, np.inf])
+def test_gh_17846(f0_fill):
+    with pytest.raises(ValueError):
+        solve_ivp(fun_zero, [0, 10], np.full(3, f0_fill))
```

Reply:
I am not sure I understand what you mean. I have compiled the branch associated with the PR, and calling `solve_ivp` with non-finite `y0` did raise a `ValueError`.